sentinel-extension/sentinel-datasource-nacos/src/main/java/com/alibaba/csp/sentinel/datasource/nacos/NacosDataSource.java

```diff
@@ -66,13 +69,26 @@
      */
     public NacosDataSource(final String serverAddr, final String groupId, final String dataId,
                            Converter<String, T> parser) {
+        this(NacosDataSource.buildProperties(serverAddr), groupId, dataId, parser);
+    }
+
+    /**
+     *
+     * @param properties properties for construct {@link ConfigService} using {@link NacosFactory#createConfigService(Properties)}
+     * @param groupId    group ID, cannot be empty
+     * @param dataId     data ID, cannot be empty
+     * @param parser     customized data parser, cannot be empty
+     */
+    public NacosDataSource(final Properties properties, final String groupId, final String dataId,
+                           Converter<String, T> parser) {
         super(parser);
-        if (StringUtil.isBlank(serverAddr) || StringUtil.isBlank(groupId) || StringUtil.isBlank(dataId)) {
-            throw new IllegalArgumentException(String.format("Bad argument: serverAddr=[%s], groupId=[%s], dataId=[%s]",
-                serverAddr, groupId, dataId));
+        if (StringUtil.isBlank(groupId) || StringUtil.isBlank(dataId)) {
+            throw new IllegalArgumentException(String.format("Bad argument: groupId=[%s], dataId=[%s]",
+                groupId, dataId));
         }
         this.groupId = groupId;
         this.dataId = dataId;
+        this.properties = properties;
```

Comment:
Here we may need to check whether the `properties` is null (using `AssertUtil.notNull(xxx)`.

Reply:
cool. nice suggestion